confmap/provider/httpprovider/provider.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpprovider // import "go.opentelemetry.io/collector/confmap/provider/httpprovider"
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+const (
+	schemeName = "http"
+)
+
+type httpClient interface {
+	Get(url string) (resp *http.Response, err error)
+}
+
+type provider struct {
+	client httpClient
+}
+
+// New returns a new confmap.Provider that reads the configuration from a file.
+//
+// This Provider supports "http" scheme, and can be called with a "uri" that follows:
+//
+// One example for http-uri be like: http://localhost:3333/getConfig
+//
+// Examples:
+// `http://localhost:3333/getConfig` - (unix, windows)
+func New() confmap.Provider {
+	return &provider{client: &http.Client{}}
+}
+
+func (fmp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
+	if !strings.HasPrefix(uri, schemeName+"://") {
```

Comment:
why do you need the authority (I think that is between the `/`) to be empty?

Reply:
Fixed, you are right. I took a look at fileprovider.